OsmAnd/src/net/osmand/plus/plugins/externalsensors/dialogs/SelectExternalDeviceFragment.java

```diff
@@ -0,0 +1,227 @@
+	protected void setupUI(@NonNull View view) {
+		super.setupUI(view);
+		emptyView = view.findViewById(R.id.empty_view);
+		contentView = view.findViewById(R.id.devices_content);
+		devicesList = view.findViewById(R.id.connected_devices_list);
+		appBar = view.findViewById(R.id.appbar);
+		noBluetoothCard = view.findViewById(R.id.no_bluetooth_card);
+
+		setupPairSensorButton(view.findViewById(R.id.pair_btn_empty));
+		setupPairSensorButton(view.findViewById(R.id.pair_btn_additional));
+		setupOpenBtSettingsButton(view.findViewById(R.id.bt_settings_button_container));
+		devicesListAdapter = new DevicesForWidgetAdapter(app, nightMode, this);
+		devicesListAdapter.setDeviceId(selectedDeviceId);
+		devicesList.setAdapter(devicesListAdapter);
+	}
+
+	private void setupPairSensorButton(@NonNull View view) {
+		View dismissButton = view.findViewById(R.id.dismiss_button);
+		int buttonTextId = R.string.ant_plus_pair_new_sensor;
+		ViewGroup.LayoutParams layoutParams = dismissButton.getLayoutParams();
+		UiUtilities.setupDialogButton(nightMode, dismissButton, UiUtilities.DialogButtonType.SECONDARY, buttonTextId);
+		layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
+		dismissButton.setLayoutParams(layoutParams);
+		view.requestLayout();
+		dismissButton.setOnClickListener(v -> showPairNewSensorBottomSheet());
+		AndroidUiHelper.updateVisibility(dismissButton, true);
+	}
+
+	private void setupOpenBtSettingsButton(@NonNull View view) {
+		View dismissButton = view.findViewById(R.id.dismiss_button);
+		int buttonTextId = R.string.ant_plus_open_settings;
+		ViewGroup.LayoutParams layoutParams = dismissButton.getLayoutParams();
+		UiUtilities.setupDialogButton(nightMode, dismissButton, UiUtilities.DialogButtonType.SECONDARY, buttonTextId);
+		layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;
+		dismissButton.setLayoutParams(layoutParams);
+		view.requestLayout();
+		dismissButton.setOnClickListener(v -> {
+			Intent intentOpenBluetoothSettings = new Intent();
+			intentOpenBluetoothSettings.setAction(Settings.ACTION_BLUETOOTH_SETTINGS);
+			startActivity(intentOpenBluetoothSettings);
+		});
+		AndroidUiHelper.updateVisibility(dismissButton, true);
+	}
+
+	private void showPairNewSensorBottomSheet() {
+		PairNewDeviceBottomSheet.showInstance(requireActivity().getSupportFragmentManager());
+	}
+
+	@Override
+	protected void setupToolbar(@NonNull View view) {
+		super.setupToolbar(view);
+		Toolbar toolbar = view.findViewById(R.id.toolbar);
+		toolbar.setOnMenuItemClickListener(item -> {
+			if (item.getItemId() == R.id.action_add) {
+				showPairNewSensorBottomSheet();
+			}
+			return false;
+		});
+	}
+
+	@Override
+	public void onDestroyView() {
+		super.onDestroyView();
+		contentView = null;
+		emptyView = null;
+		devicesList = null;
+		appBar = null;
+		noBluetoothCard = null;
+	}
+
+	@Override
+	public void onResume() {
+		super.onResume();
+		noBluetoothCard.setVisibility(plugin.isBlueToothEnabled() ? View.GONE : View.VISIBLE);
```

Comment:
If Bluetooth is disabled and devices are empty and we hide them [here](https://github.com/osmandapp/OsmAnd/pull/17242/files#diff-e8f72e7c9cf4919acfd4898f1517c24a461a05d5aaf666be28962e58387e16aaR170), noBluetoothCard will be gone as part of contentView? Is it intended behavior?

Reply:
yes. in such separate noBluetoothCard will be shown - they have different layouts